docs/requirements.pip

```diff
@@ -0,0 +1,5 @@
+mkdocs-material
```

Comment:
Please let's pin to explicit versions to avoid unexpected behavior as updated change?

Reply:
I didn't pin the explicit version (bad practice in Python), but the last major range (which is a general Python packaging recommendation). Python does not have binary compatibility concept, so it is better to be able to fetch last bug fixes or security fixes unless you have an explicit dependency on a specific version (which is not the case here). Pinning explicit versions will later lead to version conflicts (because you might pin a different version than a transitive dependency could and Python can't provide multiple versions of a dependency in the same venv)

If you want to pin all versions, we should use a package manager using a lock file (which I didn't because this is a Go project and I wanted to keep the tooling as light as possible and pip is builtin Python).

Version pinning in requirements file only pin the leaf, not the transitive dependencies (so it don't bring reproducibility to ping versions in requirements file).

Note that `mkdocs-material` is strictly following semver/PEP440 semantic to ensure you can do that. So are `pygments` and `mkdocs-macro-plugin`.

I made an exception for `mdx-breakless-lists` which only have 2 version (`1.0` and `1.0.1`) not really following semvr/pep440 and doesn't really ahve much activity

Reply:
Thanks for elaborating.

My dependencies habit is to explicitly pin, and wire dependabot/renovatebot (sometimes with auto-merge; depends on the project). That way there is an explicit choice captured with a commit to take a new upstream version, and that means an event to provide an opportunity to test it for correctness (and taint).

It's a matter of trust; dependencies can be well-behaved (as these), but also not.

Reply:
File has been updated with all dependencies pinned 👍🏼